src/pile.js

```diff
@@ -1032,6 +1032,55 @@ const createPile = (
     setCover(newCover);
   };
 
+  let labelGraphics;
+
+  const drawLabel = (labels, colors, texts) => {
+    if (!labels.length) {
+      labelGraphics.clear();
+      labelGraphics.removeChildren();
+      return;
+    }
+
+    if (isPositioning || isScaling) {
+      postPilePositionAnimation.set('drawLabel', () => {
+        drawLabel(labels, colors, texts);
+      });
+      return;
+    }
+
+    if (!labelGraphics) {
+      labelGraphics = new PIXI.Graphics();
+      contentGraphics.addChild(labelGraphics);
+    } else {
+      labelGraphics.clear();
+      labelGraphics.removeChildren();
+    }
+
+    const { width, height } = contentGraphics;
+
+    const labelWidth = width / labels.length;
+    const labelHeight = 8;
+    labels.forEach((label, index) => {
+      const labelX = labelWidth * index - width / 2;
```

Comment:
I think this position needs to be adjusted with on the `baseScale`. Maybe `labelX /= baseScale`?